zio-http/src/main/scala/zhttp/http/URL.scala

```diff
@@ -45,12 +47,22 @@ object URL                                                             {
 
     } yield url
 
-  def asString(url: URL): String = url.kind match {
-    case Location.Relative                     => url.path.asString
-    case Location.Absolute(scheme, host, port) =>
-      port match {
-        case 80   => s"${scheme.asString}://${host}${url.path.asString}"
-        case port => s"${scheme.asString}://${host}:${port}${url.path.asString}"
+  def asString(url: URL): String = {
+    def withQuery(u: String): String =
```

Comment:
can you move `withQuery` out of `asString`. This could cause perf issues.

Reply:
How does that cause perf issues?